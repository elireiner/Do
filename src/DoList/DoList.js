import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import ToDo from '../ToDo/ToDo';
import Context from '../Context'

class DoList extends Component {
    static defaultProps = {
        ToDoList: []
    }

    static contextType = Context;

    render() {

        const { ToDoList } = this.context

        return (
            <>
            <ul>
                {ToDoList.map((item) => (<ToDo name={item.name} />))}
            </ul>
            <Link to={`/AddForm`}>Add ToDo</Link>
            </>
        )
    }
}

export default DoList