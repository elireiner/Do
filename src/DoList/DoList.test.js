import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import DoList from './DoList';

describe('DoList', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<DoList />, div);
        ReactDOM.unmountComponentAtNode(div);
    })
    it('renders UI as expected', () => {
        const tree = renderer.create(<DoList />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})