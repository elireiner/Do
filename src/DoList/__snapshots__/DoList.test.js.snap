// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoList renders UI as expected 1`] = `<ul />`;
